backend/src/data/repositories/course/course.repository.ts

```diff
@@ -37,11 +43,35 @@ class Course {
           builder.where('title', 'ilike', `%${title}%`);
         }
       })
+      .innerJoin(
+        'courseCategories',
+        'courses.courseCategoryId',
+        'courseCategories.id',
+      )
       .withGraphJoined('vendor')
       .castTo<CourseGetResponseDto[]>()
       .execute();
   }
 
+  public async getAllPaginated({
```

Comment:
```suggestion
  public async getAll({
```
and add new method getAllWithCategories